src/routes/committee/setlist/+page.svelte

```diff
@@ -0,0 +1,249 @@
+			edition: doc(db, 'editions', 'ZI3Eab1mXjHvCUS47o40'),
+			part: instrument,
+			song: doc(db, 'songs', data.songs[selectedSong].id)
+		})
+		data.musiciansForSongs[data.songs[selectedSong].id] = [
+			...data.musiciansForSongs[data.songs[selectedSong].id],
+			[data.namesMap[participant], instrument]
+		]
+	}
+
+	async function RemoveParticipantFromSong() {
+		const userQuery = query(collection(db, 'users'), where('name', '==', selectedUser))
+		const userId = (await getDocs(userQuery)).docs[0].id
+		const songRef = doc(db, 'songs', data.songs[selectedSong].id)
+
+		const playsInQuery = query(
+			collection(db, 'users', userId, 'playsSongInEdition'),
+			where('song', '==', songRef),
+			where('part', '==', selectedInstrument)
+		)
+		const playsInDoc = (await getDocs(playsInQuery)).docs[0].ref
+
+		await deleteDoc(playsInDoc)
+
+		data.musiciansForSongs = data.musiciansForSongs
+	}
+</script>
+
+<Grid>
+	<Row>
+		<Column><h1 class="titleText">Setlist</h1></Column>
+	</Row>
+
+	<StructuredList condensed>
+		<StructuredListHead>
+			<StructuredListRow head>
+				<StructuredListCell head>Title</StructuredListCell>
+				<StructuredListCell head>Artist</StructuredListCell>
+				<StructuredListCell head>Link</StructuredListCell>
+				<StructuredListCell head>Line-up</StructuredListCell>
+			</StructuredListRow>
+		</StructuredListHead>
+		{#each data.songs as song, i}
+			<StructuredListRow>
+				<StructuredListCell>{song.name}</StructuredListCell>
+				<StructuredListCell>{song.artist}</StructuredListCell>
+				<StructuredListCell
+					>{@const validUrl = isValidUrl(song.link)}
+					<Button
+						href={validUrl ? song.link : undefined}
+						target="blank"
+						kind="ghost"
+						size="small"
+						iconDescription={validUrl ? song.link : 'Invalid URL'}
+						icon={PlayFilledAlt}
+						disabled={!validUrl}
+					/></StructuredListCell
+				>
+				<StructuredListCell>
+					{#each Object.entries(data.musiciansForSongs) as [key, value], j}
+						{#if key == song.id}
+							{#each value as musician}
+								{musician[0]} - {musician[1]}
+								<Button
+									size="small"
+									kind="ghost"
+									class="removePerson"
+									on:click={() => {
+										selectedSong = i
+										selectedUser = musician[0]
+										selectedInstrument = musician[1]
+										selectedUserEntry = j
+										openDelUser = true
+									}}>X</Button
+								>
+								<br />
+							{/each}
+						{/if}
+					{/each}
+				</StructuredListCell>
+				<StructuredListCell>
+					<Button
+						kind="ghost"
+						size="small"
+						iconDescription="Add musician"
+						icon={UserFollow}
+						on:click={() => {
+							selectedSong = i
+							openAdd = true
+						}}
+					/>
+					<Button
+						kind="danger-tertiary"
+						size="small"
+						iconDescription="Delete"
+						icon={MusicRemove}
+						on:click={() => {
+							selectedSong = i
+							openDel = true
+						}}
+					/>
```

Comment:
Voor dit soort dingen worden langzaam misschien ook handig te gaan kijken of we knopjes kunnen hergebruiken, maakt deze page files ook makkelijker te lezen